fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java

```diff
@@ -86,7 +86,7 @@
 
     private final PlannerContext ctx_;
     private final ArrayList<ScanNode> scanNodes = Lists.newArrayList();
-    private Map<UUID, List<ScanNode>> selectStmtToScanNodes = Maps.newHashMap();
+    private Map<Analyzer, List<ScanNode>> selectStmtToScanNodes = Maps.newHashMap();
```

Comment:
Is the Analyzer of CTE same as outer query?

Reply:
No. 
For SQL
```
select id, t1 from(
select id, a as t1 from t
union all
select id, aaa as t1 from t
) k2;
```
The analyzer instances for `select id, a as t1 from t` and `select id, aaa as t1 from t` are different.
Such as, The analyzer instances for `select id, a as t1 from t` is 6177, which ancestors are[6219,6220,6221],
The analyzer instances for `select id, aaa as t1 from t` is 6225, which ancestors are[6247,6220,6221].